ansible/roles/create-s390x-vm/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+- name: Fetch the variables from var file
+  include_vars:
+    file: vars.yml
+
+- name: Check for existing VSI
+  ibm.cloudcollection.ibm_is_instance_info:
+    name: "{{ vsi_name }}"
+  failed_when:
+    - vsi.rc != 0
+    - '"No Instance found" not in vsi.stderr'
+  register: vsi
+
+- name: Configure VSI
+  ibm.cloudcollection.ibm_is_instance:
+    name: "{{ vsi_name }}"
+    state: available
+    id: "{{ vsi.resource.id | default(omit) }}"
+    vpc: "{{ vpc_id }}"
+    profile: "{{ vsi_profile }}"
+    image: "{{ vsi_image_id }}"
+    resource_group: "{{ vsi_resource_group_id }}"
+    keys:
+      - "{{ ssh_key_id }}"
+    primary_network_interface:
+     - subnet: "{{ subnet_id }}"
+    zone: "{{ zone }}"
+  register: vsi
+
+- name: Check for existing Floating IP
+  ibm.cloudcollection.ibm_is_floating_ip_info:
+    name: "{{ vsi_name }}-fip"
+  failed_when:
+    - fip.rc != 0
+    - '"No floatingIP found" not in fip.stderr'
+  register: fip
+
+- name: Configure Floating IP Address
+  ibm.cloudcollection.ibm_is_floating_ip:
+    name: "{{ vsi_name }}-fip"
+    state: available
+    id: "{{ fip.resource.id | default(omit) }}"
+    resource_group: "{{ vsi_resource_group_id }}"
+    target: "{{ vsi.resource.primary_network_interface[0]['id'] }}"
+  register: fip
+
+- name: Print Floating IP Address
+  debug:
+    msg: "{{ vsi_name }} IP Address: {{ fip.resource.address }}"
+
+- name: Add VSI to Ansible inventory
+  add_host:
```

Comment:
How does this work with the wider inventory code? and how does this persist between executions of playbooks?

At the moment the process tends to be to create all the VMs first, then in a subsequent execution we run the tests (and a final step destroys the VMs.)

Reply:
I did test the vm-lifecycle playbook (i.e., creation, provision, destroy) with the `add_host` approach, and it was working ok. I expect the integration test (i.e,. vm creation, vm provision, test, vm destroy) should be ok, as long as they are in the same playbook. But this hasn't been tested yet since we don't have test environment ready (e.g., no multiarch image for building or testing).

Reply:
After reading the workflow, I agree the inventory management is needed. Therefore, adding this commite https://github.com/stackrox/collector/pull/1249/commits/cebcaa94059e25defcc151ff6f4ed73e0c5c2bd2

Reply:
Looks good! One minor point, could we add `ansible/{dev,ci}/inventory_ibmcloud_z.yml` to `.gitignore` ? It's unlikely that it'll be committed but worth enforcing I think.

Reply:
Done!